encoder.go

```diff
@@ -123,22 +140,33 @@ func EncodeTime(w io.Writer, t time.Time) (err error) {
 	// TODO: We are hand-encoding a struct until MarshalAmino/UnmarshalAmino is supported.
 	// skip if default/zero value:
 	if s != 0 {
-		err = encodeFieldNumberAndTyp3(w, 1, Typ3_8Byte)
+		if s < minSeconds || s >= maxSeconds {
+			return InvalidTimeErr(fmt.Sprintf("seconds have to be > %d and < %d, got: %d",
+				minSeconds, maxSeconds, s))
+		}
+		err = encodeFieldNumberAndTyp3(w, 1, Typ3_Varint)
 		if err != nil {
 			return
 		}
-		err = EncodeInt64(w, s)
+		err = EncodeUvarint(w, uint64(s))
 		if err != nil {
 			return
 		}
 	}
 	// skip if default/zero value:
 	if ns != 0 {
-		err = encodeFieldNumberAndTyp3(w, 2, Typ3_4Byte)
+		// do not encode if not in interval [0, 999999999]
+		if ns < 0 || ns > maxNanos {
+			// we could as well panic here:
+			// time.Time.Nanosecond() guarantees nanos to be in [0, 999,999,999]
+			return InvalidTimeErr(fmt.Sprintf("nanosecons have to be >= 0 and <= %v, got: %d",
```

Comment:
`nanoseconds`

Reply:
Thanks. I'll address this and above comment in #233